cmd/desync/progressbar.go

```diff
@@ -39,3 +39,48 @@ func (p ProgressBar) Start() {
 func (p ProgressBar) Set(current int) {
 	p.ProgressBar.Set(current)
 }
+
+// ProgressReader wraps ProgressBar and implements io.Reader
+type ProgressReader struct {
+	desync.ProgressBar
+	f *os.File
+}
+
+func NewProgressReader(prefix string, f *os.File) *ProgressReader {
+	pb := NewProgressBar(prefix)
+	if pb == nil {
+		return nil
+	}
+
+	cur, err := f.Seek(0, os.SEEK_CUR)
+	if err != nil {
+		return nil
+	}
+
+	size, err := f.Seek(0, os.SEEK_END)
+	f.Seek(cur, os.SEEK_SET)
+	if err != nil {
+		return nil
+	}
+
+	pr := ProgressReader{pb, f}
+	pb.SetTotal(int(size))
+	pb.Set(int(cur))
+
+	return &pr
+}
+
+func (p ProgressReader) Read(b []byte) (n int, err error) {
+	n, err = p.f.Read(b)
+	if err != nil {
+		return
+	}
+
+	cur, err := p.f.Seek(0, os.SEEK_CUR)
```

Comment:
This could be handled by incrementing the progress bar by `n` rather than the seek